pkg/cmd/extension/build/cmd_test.go

```diff
@@ -15,186 +15,162 @@
-	"os"
-	"path/filepath"
-	"strings"
+	"os/user"
+	"testing"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
 
-	"github.com/tetratelabs/getenvoy/pkg/cmd"
-	testcontext "github.com/tetratelabs/getenvoy/pkg/test/cmd/extension"
+	"github.com/tetratelabs/getenvoy/pkg/test/cmd/extension"
 	cmdutil "github.com/tetratelabs/getenvoy/pkg/util/cmd"
 )
 
-var _ = Describe("getenvoy extension build", func() {
+// relativeWorkspaceDir points to a usable pre-initialized workspace
+const relativeWorkspaceDir = "testdata/workspace"
 
-	var dockerDir string
-
-	BeforeEach(func() {
-		dir, err := filepath.Abs("../../../extension/workspace/toolchain/builtin/testdata/toolchain")
-		Expect(err).ToNot(HaveOccurred())
-		dockerDir = dir
-	})
-
-	var pathBackup string
-
-	BeforeEach(func() {
-		pathBackup = os.Getenv("PATH")
-
-		// override PATH to overshadow `docker` executable during the test
-		path := strings.Join([]string{dockerDir, pathBackup}, string(filepath.ListSeparator))
-		os.Setenv("PATH", path)
-	})
-
-	AfterEach(func() {
-		os.Setenv("PATH", pathBackup)
-	})
-
-	var cwdBackup string
-
-	BeforeEach(func() {
-		cwd, err := os.Getwd()
-		Expect(err).ToNot(HaveOccurred())
-		cwdBackup = cwd
-	})
-
-	AfterEach(func() {
-		if cwdBackup != "" {
-			Expect(os.Chdir(cwdBackup)).To(Succeed())
-		}
-	})
-
-	testcontext.SetDefaultUser() // UID:GID == 1001:1002
-
-	var stdout *bytes.Buffer
-	var stderr *bytes.Buffer
-
-	BeforeEach(func() {
-		stdout = new(bytes.Buffer)
-		stderr = new(bytes.Buffer)
-	})
-
-	var c *cobra.Command
-
-	BeforeEach(func() {
-		c = cmd.NewRoot()
-		c.SetOut(stdout)
-		c.SetErr(stderr)
-	})
-
-	It("should validate value of --toolchain-container-image flag", func() {
-		By("running command")
-		c.SetArgs([]string{"extension", "build", "--toolchain-container-image", "?invalid value?"})
-		err := cmdutil.Execute(c)
-		Expect(err).To(HaveOccurred())
-
-		By("verifying command output")
-		Expect(stdout.String()).To(BeEmpty())
-		Expect(stderr.String()).To(Equal(`Error: "?invalid value?" is not a valid image name: invalid reference format
-
-Run 'getenvoy extension build --help' for usage.
-`))
-	})
-
-	It("should validate value of --toolchain-container-options flag", func() {
-		By("running command")
-		c.SetArgs([]string{"extension", "build", "--toolchain-container-options", "imbalanced ' quotes"})
-		err := cmdutil.Execute(c)
-		Expect(err).To(HaveOccurred())
-
-		By("verifying command output")
-		Expect(stdout.String()).To(BeEmpty())
-		Expect(stderr.String()).To(Equal(`Error: "imbalanced ' quotes" is not a valid command line string
-
-Run 'getenvoy extension build --help' for usage.
-`))
-	})
-
-	chdir := func(path string) string {
-		dir, err := filepath.Abs(path)
-		Expect(err).ToNot(HaveOccurred())
-
-		err = os.Chdir(dir)
-		Expect(err).ToNot(HaveOccurred())
-
-		return dir
+func TestGetEnvoyExtensionBuildValidateFlag(t *testing.T) {
+	type TestCase struct {
```

Comment:
guess we don't need to export the inline type :smile:

Reply:
good eye!